src/NzbDrone.Core/DecisionEngine/Specifications/CutoffSpecification.cs

```diff
@@ -21,17 +30,29 @@ public CutoffSpecification(UpgradableSpecification qualityUpgradableSpecificatio
         public virtual Decision IsSatisfiedBy(RemoteMovie subject, SearchCriteriaBase searchCriteria)
         {
             var profile = subject.Movie.Profile;
+            var file = subject.Movie.MovieFile;
 
-            if (subject.Movie.MovieFile != null)
+            if (file != null)
             {
                 if (!_qualityUpgradableSpecification.CutoffNotMet(profile,
-                                                                  subject.Movie.MovieFile.Quality,
+                                                                  file.Quality,
                                                                   subject.ParsedMovieInfo.Quality))
                 {
-                    var qualityCutoffIndex = profile.GetIndex(profile.Cutoff);
-                    var qualityCutoff = profile.Items[qualityCutoffIndex.Index];
+                    file.Movie = subject.Movie;
+                    var customFormats = _formatService.ParseCustomFormat(file);
+                    var cutoff = new List<CustomFormat> { profile.FormatItems.Single(x => x.Format.Id == profile.FormatCutoff).Format };
```

Comment:
Why not move this logic into upgradableSpec so that `CutoffNotMet` returns for quality or formats? Similar to what you did with `IsUpgradable`

Reply:
The big thing here is that `CutoffNotMet` is called for all movies on initial load which I think will be too expensive.  But I take your point, a separate version `QualityCutoffNotMet` and `CutoffNotMet` might be the way forward so the logic is centralized.